### PR TITLE
feat: Refactor core commands

### PR DESCRIPTION
**Pull request by roneli**
Refactor core commands:

- [x]  Download
- [x] Downgrade / Upgrade / BuildProviderTables -> Sync
- [x] Drop 
- [x] Test
- [x] CheckForProviderUpdates
- [x] Fetch
- [x] RunPolicy
- [x] RunModule
- [x] GetProviderConfig
- [x] Refactor init
- [x] Rename client -> core and remove all calls to client
- [ ] Return analytics -> next PR
- [x] Remove client
- [x] Cleanup code and verify changes
- [x] Add console sanity tests
    - [x] Describe, Policy, Download, Sync etc'
- [x] Fix update showing when version has no update

### CODE REVIEW

File: pkg/client/fetch.go [hunk 1]
```diff
@@ -1 +1,265 @@
...
+	fetchId, err := uuid.NewUUID()
+	if err != nil {
+		return nil, diag.FromError(err, diag.INTERNAL)
+	}
+	// set metadata we want to pass to
+	metadata := map[string]interface{}{"cq_fetch_id": fetchId.String()}
+	if opts.History != nil {
+		fd := opts.History.FetchDate()
+		log.Info().Str("fetch_date", fd.Format(time.RFC3339)).Stringer("fetch_id", fetchId).Msg("history enabled adding fetch date")
+		metadata["cq_fetch_data"] = fd
```
Comment (disq) [inline pkg/client/fetch.go:81]: I would keep the comment about the extrafield cq_fetch_date from the original (also typo: "data" vs "date")

Comment (roneli) [reply pkg/client/fetch.go]: fixed

**Comment by disq** [comment]
`provider drop` is missing `--force`

**Comment by roneli** [comment]
> `provider drop` is missing `--force`

I know, will add once I merge and rebase

**Pull request merged**